plugins/settings-sync/src/com/intellij/settingsSync/SettingsSyncFiltering.kt

```diff
@@ -82,18 +78,30 @@ private fun getSchemeCategory(fileSpec: String): SettingsCategory? {
       settingsCategory = it.getSettingsCategory()
     }
   }
-  return settingsCategory
-}
 
-fun getFileSpec(path: String): String {
-  return removeOsPrefix(path)
+  if (settingsCategory == null) {
+    return null
+  }
+
+  return settingsCategory!! to null
 }
 
-private fun getSubCategory(fileSpec: String): String? {
-  if (fileSpec == AppEditorFontOptions.STORAGE_NAME)
-    return EDITOR_FONT_SUBCATEGORY_ID
-  else
-    return null
+private fun getSubCategory(fileSpec: String, category: SettingsCategory, componentClass: Class<PersistentStateComponent<Any>>): String? {
+  // NOTE: we do not cache the subcategory here, as it's cached in the `categoryCache`
+  val subcategory = when (category) {
+    SettingsCategory.UI ->
+      if (fileSpec == AppEditorFontOptions.STORAGE_NAME) EDITOR_FONT_SUBCATEGORY_ID else null
+    SettingsCategory.PLUGINS -> {
```

Comment:
That PLUGINS category is a bit tricky. It isn't in fact used as it is written here, but rather very rarely as a category (see `com.intellij.openapi.updateSettings.impl.pluginsAdvertisement.GlobalIgnoredPluginSuggestionState` as an example), so I will revert the method to the original one.

Reply:
Hmmm, it's been a while since I wrote this so maybe I misremember it, but as far as I could tell without this change you can't really disable syncing for a single plugin (both sync code and apply code consult `isSyncCategoryEnabled` as far as I could tell). This makes the `Configure` dropdown by the `Plugins` checkbox end up being dead and misleading UI — no matter what you choose there, you can only either sync all plugins or none.

If you prefer to keep this broken behaviour for now and fix it separately then sure, we can do it like that. Could you just clarify what exactly you want me to do here? You gave an approve and said "I will revert" — do I have to do something there, or does that mean you will finish this PR yourself?

Reply:
@jaen indeed, there's some complication and inconsistency here, but I'd like to address this separately

Reply:
>  You gave an approve and said "I will revert" — do I have to do something there, or does that mean you will finish this PR yourself?
nope, I did this myself